cmd/snap-assert/main.go

```diff
@@ -0,0 +1,123 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/asserts/tool"
+)
+
+func main() {
+	if err := run(); err != nil {
+		fmt.Fprintf(os.Stderr, "snap-assert: %s\n", err)
+		os.Exit(1)
+	}
+}
+
+func run() error {
+	var opts struct {
+		Positional struct {
+			AssertionType string `positional-arg-name:"<assert-type>" required:"yes" description:"type of the assertion to sign (mandatory)"`
+			Statement     string `positional-arg-name:"<statement>" description:"input file with the statement to sign as YAML or JSON (optional, left out or - means use stdin)"`
+		} `positional-args:"yes"`
+
+		Format string `long:"format" default:"yaml" description:"the format of the input statement (json|yaml)"`
+
+		AuthorityID string `long:"authority-id" description:"identifier of the signer (otherwise taken from the account-key or the statement)"`
+		KeyID       string `long:"key-id" description:"long key id of the GnuPG key to use (otherwise taken from account-key)"`
+		AccountKey  string `long:"account-key" description:"file with the account-key assertion of the key to use"`
+
+		Revision int `long:"revision" description:"revision to set for the assertion (starts and defaults to 0)"`
```

Comment:
it's a bit true for the lot of them, I think we'll want some actual document about this at some point soon